src/server/svr_jobfunc.c

```diff
@@ -1470,6 +1471,111 @@ svr_chkque(job *pjob, pbs_queue *pque, char *hostname, int mtype)
 	return (0);	/* all ok, job can enter queue */
 }
 
+/**
+ * @brief
+ *		check_block_wt	-	A worktask to reply to the blocked job client
+ *
+ * @param[in]	ptask	-	work_task structure
+ */
+void
+check_block_wt(struct work_task *ptask) {
+	
+	struct block_job_reply	*blockj = ptask->wt_parm1;
+	int time_difference;
+	time_t current_timestamp = time(NULL);
+	int is_connected = *(int *)ptask->wt_parm2;
+	int is_replied = 0;
+	int is_error = 0;
+
+	if (is_connected != 1) {
```

Comment:
you can put the connect() also in this function itself, so that we can retry even if connect itself fails